File: Bulgarian Competitions/BNOI - Group D/Lottery/Lottery.cs
using System;
using System.Linq;

class Lottery
{
    static void Main()
    {
        int k = Console.ReadLine().Split(' ').Select(m => int.Parse(m)).ToArray()[1];
        int[] A = Console.ReadLine().Split(' ').Select(m => int.Parse(m)).ToArray();
        int[] MinMax = Console.ReadLine().Split(' ').Select(m => int.Parse(m)).ToArray();
        int min = MinMax[0];
        int len = MinMax[1] - min + 1;
        int[] result = new int[len];
        for (int a = 0; a < A.Length; a++)
        {
            int start = ((min % A[a] == 0) ? 0 : A[a] - min % A[a]);
            for (int i = start; i < len; i += A[a])
            {
                result[i]++;
            }
        }

        Console.WriteLine(result.Where(m => m == k).Count());
    }
}

File: Bulgarian Competitions/BNOI - Group D/Lottery/Lottery.js
function solve(str) {
    var args = str[0].split('\n');
    var k = args[0].split(' ')[1];
    var arr = args[1].split(' ');
    var minMax = args[2].split(' ');
    var min = minMax[0];
    var len = minMax[1] - min + 1;
    var result = [];
    for (var a = 0; a < arr.length; a++) {
        var start = ((min % arr[a] == 0) ? 0 : arr[a] - min % arr[a]);
        for (var i = start; i < len; i += +arr[a]) {
            result[i] = (result[i]) ? result[i] + 1 : 1;
        }
    }

    return result.filter(function (m) {
        return m == k;
    }).length;
}